ddl/ddl_worker.go

```diff
@@ -485,6 +485,53 @@ func (w *worker) updateDDLJob(t *meta.Meta, job *model.Job, meetErr bool) error
 	return errors.Trace(err)
 }
 
+// registerMDLInfo registers metadata lock info.
+func (w *worker) registerMDLInfo(job *model.Job, ver int64) error {
+	if !variable.EnableMDL.Load() {
+		return nil
+	}
+	if ver == 0 {
+		return nil
+	}
+	rows, err := w.sess.execute(context.Background(), fmt.Sprintf("select table_ids from mysql.tidb_ddl_job where job_id = %d", job.ID), "register-mdl-info")
+	if err != nil {
+		return err
+	}
+	if len(rows) == 0 {
+		return errors.Errorf("can't find ddl job %d", job.ID)
+	}
+	ids := rows[0].GetString(0)
+	sql := fmt.Sprintf("replace into mysql.tidb_mdl_info (job_id, version, table_ids) values (%d, %d, '%s')", job.ID, ver, ids)
+	_, err = w.sess.execute(context.Background(), sql, "register-mdl-info")
+	return err
+}
+
+// cleanMDLInfo cleans metadata lock info.
+func cleanMDLInfo(pool *sessionPool, jobID int64) {
+	if !variable.EnableMDL.Load() {
+		return
+	}
+	sql := fmt.Sprintf("delete from mysql.tidb_mdl_info where job_id = %d", jobID)
+	sctx, _ := pool.get()
+	defer pool.put(sctx)
+	sess := newSession(sctx)
+	sess.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
+	_, _ = sess.execute(context.Background(), sql, "delete-mdl-info")
```

Comment:
We should at least print a log on error, so that the problem could be discovered in time.